### PR TITLE
[zh-TW]Translate String.prototype.localeCompare()

### PR DESCRIPTION
**Pull request by shiyou0130011**
as the title said, to translate `String.prototype.localeCompare()`

### CODE REVIEW

File: files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+title: String.prototype.localeCompare()
+slug: Web/JavaScript/Reference/Global_Objects/String/localeCompare
+tags:
+  - Internationalization
+  - JavaScript
+  - String.prototype.localeCompare()
+  - Prototype
+  - 参考
+  - 字串
+translation_of: Web/JavaScript/Reference/Global_Objects/String/localeCompare
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:11]: see #7412

```suggestion
```

File: files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+---
+{{JSRef}}
+
+**`localeCompare()`** 回傳一個數字，用來表示其與被比較的字串的先後順序。
+
+如果環境中有支援[`Intl.Collator` API](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator)，這個方法實際上是調用 `Intl.Collator` API。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:17]: follow https://github.com/mdn/translated-content/blob/main/docs/zh-tw/translation-guide.md#%E7%AB%99%E5%85%A7%E9%80%A3%E7%B5%90%E7%BF%BB%E8%AD%AF

```suggestion
如果環境中有支援[`Intl.Collator` API](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator)，這個方法實際上是調用 `Intl.Collator` API。
```

**Review by yin1999 — DISMISSED** [review]

File: files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md (hunk 2 above)
Comment (yin1999) [reply files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md]: There are a lot of links needed to be updated, please follow this one.

File: files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+localeCompare(compareString, locales)
+localeCompare(compareString, locales, options)
+```
+
+### 參數
+
+`locales` 和 `options` 參數可以調整函數的回傳結果，並且能指定要依照哪種語言來進行比較。
+
+在實現 [`Intl.Collator` API](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator)的環境中，這些參數與 [`Intl.Collator()`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator) 的參數互相對應；
+若環境沒有實現 `Intl.Collator` ，則這兩個參數就會被忽略，其回傳結果完全看該環境是如何實現此方法的。唯一能確定回傳結果的情況是比較的結果相等時。
```
Comment (irvin) [inline files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:26]: > making the comparison result returned entirely implementation-dependent — it's only required to be consistent.

其回傳結果完全看該環境如何實現此方法，而唯一的要求是必須維持一致。

**Review by irvin — CHANGES_REQUESTED** [review]
我覺得這邊的意思應該是說「平台可以自行決定如何比較，但必須在任何 case 下都維持一致」？ @yin1999 你覺得呢？

File: files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md (hunk 3 above)
Comment (yin1999) [reply files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:26]: 参考规范：

> The actual return values are [implementation-defined](https://tc39.es/ecma262/multipage/overview.html#implementation-defined) to permit encoding additional information in them, but this method, when considered as a method of two arguments, is required to be a [consistent comparator](https://tc39.es/ecma262/multipage/indexed-collections.html#consistent-comparator) defining a total ordering on the set of all Strings. This method is also required to recognize and honour canonical equivalence according to the Unicode Standard, including returning +0𝔽 when comparing distinguishable Strings that are canonically equivalent.

我认为最后一句话的意思应该是指，对于同样的两个字符串，其比较的结果应该始终一致（the same value `V`）

image

Comment (irvin) [reply files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:26]: 其回傳結果完全看該環境如何實作，唯一的要求是（對於同樣的字串）比較結果必須始終一致。

Comment (irvin) [inline files/zh-tw/web/javascript/reference/global_objects/string/localecompare/index.md:26]:
```suggestion
若環境沒有實現 `Intl.Collator` ，則這兩個參數就會被忽略，其回傳結果完全看該環境如何實作，唯一的要求是（對於同樣的字串）比較結果必須始終一致。
```

**Review by irvin — CHANGES_REQUESTED** [review]

**Pull request merged**